### DISCUSSION TITLE
Is there a way to implement a weighted average using the groupby function?

### DISCUSSION THREAD

**Discussion by franlane94** [h2oai/datatable#2969] [Q&A] [answered]
Hi all,

I was wondering whether there is a neat way to implement a weighted mean when using the groupby function. I would achieve this in Pandas by applying a function when grouping. However, I can't see a way to do this using datatable.

Thanks!

**Comment by st-pasha** [comment, marked as answer]
Formula `dt.sum(f.X * f.weight) / dt.sum(f.weight)` will compute weighted average of a column `X`, and you can use it either with or without a groupby.
